shiny/reactive/_core.py

```diff
@@ -327,12 +333,19 @@ def invalidate_later(
     run.
     """
 
+    cur_session: Session | MISSING_TYPE | None = None
+
     if isinstance(session, MISSING_TYPE):
         from ..session import get_current_session
 
         # If no session is provided, autodetect the current session (this
         # could be None if outside of a session).
-        session = get_current_session()
+        cur_session = get_current_session()
+    else:
+        # Display warning that they should use `session_context()`
+        session_type_warning()
+        # Use current session value
+        cur_session = session
```

Comment:
This is the only section that did not use `active_session = require_active_session()` cadence as an active session isn't required given the code. 

I'd argue a session should be required to call `invalidate_later(delay)`